icu4c/source/test/intltest/listformattertest.cpp

```diff
@@ -158,7 +211,231 @@ void ListFormatterTest::TestEnglishGB() {
         one + ", " + two + ", " + three + " and " + four
     };
 
-    CheckFourCases("en_GB", one, two, three, four, results);
+    CheckFourCases("en_GB", one, two, three, four, results, "TestEnglishGB()");
+}
+
+void ListFormatterTest::TestFieldPositionIteratorWontCrash() {
+    IcuTestErrorCode errorCode(*this, "TestFieldPositionIteratorWontCrash()");
```

Comment:
Here and elsewhere: consider the other methods in IcuTestErrorCode.

```
    // Returns TRUE if isFailure().
    UBool errIfFailureAndReset();
    UBool errIfFailureAndReset(const char *fmt, ...);
    UBool errDataIfFailureAndReset();
    UBool errDataIfFailureAndReset(const char *fmt, ...);
```

This lets you do

```
if (errorCode.errDataIfFailureAndReset()) {
    return;
}
```

It gives you a message containing the test function name.  You can add more to the message using one of the format string overloads or by using `setScope()`.

Reply:
ack